File: eva-core/eva-core-common/src/main/java/io/nerv/core/util/OrderHelper.java
package io.nerv.core.util;

import java.util.Random;

/**
 * 订单帮助类
 *
 * @author: S.PKAQ
 */
public class OrderHelper {
    private final static String prefix = "O";

    /**
     * 生成订单号
     *
     * @param userCode
     * @return
     */
    public static String getOrderNumber(String userCode) {
        long timestamp = System.currentTimeMillis();
        StringBuilder sb = new StringBuilder();
        sb.append(prefix);
        sb.append(timestamp);
        sb.append(userCode);
        sb.append(new Random().nextInt());
        return sb.toString();
    }
}

File: eva-core/src/main/kotlin/io/nerv/core/util/OrderHelper.kt
package io.nerv.core.util

import java.util.*

/**
 * 订单帮助类
 * @author: S.PKAQ
 * @Datetime: 2018/3/26 23:52
 */
object OrderHelper {
    private const val prefix = "O"

    /**
     * 生成订单号
     * @param userCode
     * @return
     */
    fun getOrderNumber(userCode: String?): String {
        val timestamp = System.currentTimeMillis()
        val sb = StringBuilder()
        sb.append(prefix)
        sb.append(timestamp)
        sb.append(userCode)
        sb.append(Random().nextInt())
        return sb.toString()
    }
}